File: Python/1-SRP/SRP-Solucao/ClienteService.py
from ClienteRepository import ClienteRepository

class ClienteService:
    def adicionarCliente(self, cliente):
        if (not cliente.isValid()):
            return 'Dados inválidos'
        
        ClienteRepository().adicionarCliente(cliente)
        
        return 'ClienteService: Cliente cadastrado com sucesso \n'

File: PHP/1-SRP/SRP-Solucao/ClienteService.php
<?php
require_once('EmailServices.php');
require_once('ClienteRepository.php');

class ClienteService
{
    function adicionarCliente($cliente)
    {
        if (!$cliente->isValid())
            return "Dados inválidos";

        $repo = new ClienteRepository();
        $repo->adicionarCliente($cliente);
        
        $emailServices = new EmailServices();
        $emailServices->enviar("empresa@empresa.com", $cliente->nome, $cliente->email, "Bem Vindo", "Parabéns está Cadastrado");

        return "ClienteService: Cliente cadastrado com sucesso \n";
    }
}
